src/pebble.py

```diff
@@ -38,31 +41,25 @@ def __init__(self, msg: str):
         self.msg = msg
 
 
-def check_synapse_ready() -> ops.pebble.CheckDict:
+def check_synapse_ready(charm_state: CharmState) -> ops.pebble.CheckDict:
```

Comment:
do you think this would need the entire `charm_state` instance as parameter? can we only pass the `ready` config?

Reply:
The charm is not using this pattern yet so I follow the existing one, at least until we have a refactor in place to change that.
Also, I could check if is None or not in check_synapse_ready